internal/core/command/put_test.go

```diff
@@ -48,28 +53,58 @@ func TestNewPutCommandInvalidBaseUrl(t *testing.T) {
 	device := testDevice
 	device.Service.Addressable.Address = "!@#$"
 
-	_, err := NewPutCommand(device, testCommand, "Test body", context.Background(), nil, logger.NewMockClient())
+	req := newRequestForPutWithContentType(clients.ContentTypeCBOR)
+	_, err := NewPutCommand(device, testCommand, "Test body", context.Background(), nil, logger.NewMockClient(), req)
 	if err != nil {
 		t.Errorf("The invalid URL error was not properly propegated to the caller")
 	}
 }
 
 func TestNewPutCommandBody(t *testing.T) {
 	expectedRequestBody := "Test Request Body"
-	expectedRequestBodySize := len(expectedRequestBody)
-	putCommand, err := NewPutCommand(testDevice, testCommand, expectedRequestBody, context.Background(), nil, logger.NewMockClient())
+	req := newRequestForPutWithContentType(clients.ContentTypeJSON)
+	putCommand, err := NewPutCommand(testDevice, testCommand, expectedRequestBody, context.Background(), nil, logger.NewMockClient(), req)
 
 	if err != nil {
 		t.Errorf("Unexpectedly failed while creating a PutCommand")
 	}
 
+	expectedRequestBodySize := len(expectedRequestBody)
 	actualBodyBytes, _ := ioutil.ReadAll(putCommand.(serviceCommand).Body)
 	if expectedRequestBodySize != len(actualBodyBytes) {
-		t.Errorf("Failed to verify the request body size")
+		t.Errorf("Failed to verify the req body size")
 	}
 
 	actualRequestBody := string(actualBodyBytes)
 	if expectedRequestBody != actualRequestBody {
-		t.Error("Failed to verify the request body contents")
+		t.Error("Failed to verify the req body contents")
+	}
+}
+
+func TestNewPutCommandCborContentType(t *testing.T) {
+	req := newRequestForPutWithContentType(clients.ContentTypeCBOR)
+	putCommand, _ := NewPutCommand(testDevice, testCommand, "Test Body", context.Background(), nil, logger.NewMockClient(), req)
+	actualContentType := putCommand.(serviceCommand).Request.Header.Get(clients.ContentType)
+
+	expectedContentType := clients.ContentTypeCBOR
+	if actualContentType != expectedContentType {
```

Comment:
Can we leverage the `require` package and use `require.Equal`. The test failure message is nicer and the code is simpler. This also applied to other test functions we are changing in this PR to keep the function consistent. I would say just keep these changes to the functions we are changing and not to change this everywhere as that would be a big effort. I would like to start leveraging the `require` and `assert` packages to have better consistency within the project, and better failure messages.